sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -1422,7 +1422,9 @@ def test_xfail_count_with_no_fast_fail():
             expected_failed_checks=expected_failed_checks,
             on_fail="warn",
         )
-    xfail_warns = [w for w in records if w.category != SkipTestWarning]
+    xfail_warns = [
+        w for w in records if w.category not in (SkipTestWarning, UserWarning)
```

Comment:
I prefer to update our common tests to use DataFrames with string columns and not trigger a `UserWarning`.